File: Data Structures & Algorithms/anagram-groups/submission-3.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashmap = {}

        for i in range(len(strs)):
            key = ''.join(sorted(strs[i]))
            if key not in hashmap:
                hashmap[key] = []

            hashmap[key].append(strs[i])   

        return list(hashmap.values())     



        

File: Data Structures & Algorithms/longest-common-prefix/submission-3.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix = "";

        for(int i = 0; i<strs[0].size(); i++){
            char c = strs[0][i];

            for(int s = 0; s<strs.size(); s++){
                if(i >= strs[s].size() || strs[s][i] != c){
                    return prefix;
                }
            }
            prefix += c;

        }
        return prefix;
    }
};